src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -0,0 +1,180 @@
+        field_name, content = self.vespa_chunks[chunk_index].split("::", 2)
+        return {field_name: content}
+
+
+class UnstructuredIndexDocument(BaseModel):
+    """A helper class to handle the conversion between Vespa and Marqo documents for an unstructured index.
+
+    The object can be instantiated from a Marqo document using the from_marqo_document method,
+    or can be instantiated from a Vespa document using the from_vespa_document method.
+    """
+    id: str = Field(alias=index_constants.VESPA_DOC_ID)
+    fields: UnstructuredVespaDocumentFields = Field(default_factory=UnstructuredVespaDocumentFields,
+                                                    alias=index_constants.VESPA_DOC_FIELDS)
+
+    class Config:
+        allow_population_by_field_name = True
+
+    @classmethod
+    def from_vespa_document(cls, document: Dict) -> "UnstructuredIndexDocument":
+        """Instantiate an UnstructuredIndexDocument from a Vespa document."""
+        fields = document.get(index_constants.VESPA_DOC_FIELDS, {})
+        return cls(id=document[index_constants.VESPA_DOC_ID],
+                   fields=UnstructuredVespaDocumentFields(**fields))
+
+    @classmethod
+    def from_marqo_document(cls, document: Dict) -> "UnstructuredIndexDocument":
+        """Instantiate an UnstructuredIndexDocument from a valid Marqo document from
+        add_documents"""
+
+        if index_constants.MARQO_DOC_ID not in document:
+            raise ValueError(f"Unstructured Marqo document does not have a {index_constants.MARQO_DOC_ID} field. "
+                             f"This should be assigned for a valid document")
+
+        copied = deepcopy(document)
+        doc_id = copied[index_constants.MARQO_DOC_ID]
+        instance = cls(id=doc_id, fields=UnstructuredVespaDocumentFields(marqo__id=doc_id))
+        del copied[index_constants.MARQO_DOC_ID]
+
+        for key, value in copied.items():
+            if key in [index_constants.MARQO_DOC_EMBEDDINGS, index_constants.MARQO_DOC_CHUNKS,
+                       unstructured_common.MARQO_DOC_MULTIMODAL_PARAMS]:
+                continue
+            if isinstance(value, str):
+                if len(value) <= unstructured_constants.SHORT_STRING_THRESHOLD:
+                    instance.fields.short_string_fields[key] = value
+                else:
+                    instance.fields.long_string_fields[key] = value
+                instance.fields.strings.append(value)
+            elif isinstance(value, bool):
+                instance.fields.bool_fields[key] = int(value)
+            elif isinstance(value, list) and all(isinstance(elem, str) for elem in value):
+                instance.fields.string_arrays.extend([f"{key}::{element}" for element in value])
+            elif isinstance(value, int):
+                instance.fields.int_fields[key] = value
+                instance.fields.score_modifiers_fields[key] = value
+            elif isinstance(value, float):
```

Comment:
It's always good to have a final `else:` and error out (parsing error here), in case you get unexpected input. Makes potential bugs in upstream code much easier to catch